clients/common/utility.cpp

```diff
@@ -156,6 +158,8 @@ bool match_test_category(const char* category,
                 break;
             }
         }
+
+        strcpy(arg.category, test_category);
```

Comment:
Good catch, will change it to:
```
memmove(arg.category, test_category, strlen(test_category) + 1);
```
`test_category` is either `"known_bug"` or the original `arg.category` with the `known_bug_platforms_` prefix removed, so it will never exceed the size of `arg.category` (the `rocblas_gentest.py` script will error if a string larger than `category` is specified in YAML). Most of the time the suffix is shorter than the `known_bug_platforms_` prefix so they won't overlap, but in case the suffix is longer than the `known_bug_platforms_` prefix, `memmove` is necessary.

Reply:
There is no longer need to do the `memmove`.

Now, the `category` is either left unchanged, or it is set to `known_bug`.